runtime/account_test.go

```diff
@@ -1058,92 +1060,95 @@ func TestRuntimePublicKey(t *testing.T) {
             }
         `
 
-		runtimeInterface := &testRuntimeInterface{
-			validatePublicKey: func(publicKey *PublicKey) (bool, error) {
-				return true, nil
-			},
-		}
+		runtimeInterface := &testRuntimeInterface{}
 
 		_, err := executeScript(script, runtimeInterface)
 		require.Error(t, err)
 		assert.Contains(t, err.Error(), "value of type `PublicKey` has no member `validate`")
 	})
 
-	t.Run("IsValid", func(t *testing.T) {
-		for _, validity := range []bool{true, false} {
-			script := `
-              pub fun main(): Bool {
+	t.Run("Construct PublicKey in Cadence code", func(t *testing.T) {
+		script := `
+              pub fun main(): PublicKey {
                   let publicKey = PublicKey(
                       publicKey: "0102".decodeHex(),
                       signatureAlgorithm: SignatureAlgorithm.ECDSA_P256
                   )
 
-                  return publicKey.isValid
+                  return publicKey
               }
             `
+
+		fakeError := &fakeError{}
+		for _, errorToReturn := range []error{fakeError, nil} {
 			invoked := false
-			validateMethodReturnValue := validity
 
 			storage := newTestLedger(nil, nil)
 
+
 			runtimeInterface := &testRuntimeInterface{
 				storage: storage,
-				validatePublicKey: func(publicKey *PublicKey) (bool, error) {
+				validatePublicKey: func(publicKey *PublicKey) error {
 					invoked = true
-					return validateMethodReturnValue, nil
+					return errorToReturn
 				},
 			}
 
 			value, err := executeScript(script, runtimeInterface)
-			require.NoError(t, err)
 
-			assert.True(t, invoked)
-			assert.Equal(t, cadence.Bool(validateMethodReturnValue), value)
+			assert.True(t, invoked, "validatePublicKey was not invoked")
+
+			if errorToReturn == nil {
+				assert.NotNil(t, value)
+				assert.NoError(t, err)
+			} else {
+				assert.Nil(t, value)
```

Comment:
Maybe add an assertion for the error here, e.g. `err` is the same as `errorToReturn`